src/State/Hub.php

```diff
@@ -231,21 +234,35 @@ public function startTransaction(TransactionContext $context): Transaction
     {
         $client = $this->getClient();
 
+        $sampleRate = null;
+
+        if (null !== $client) {
+            $sampler = $client->getOptions()->getTracesSampler();
+            if (null !== $sampler) {
+                if (\is_callable($sampler)) {
+                    $sampleRate = $sampler(SamplingContext::getDefault($context));
+                } elseif ($sampler instanceof TracesSamplerInterface) {
+                    $sampleRate = $sampler->sample(SamplingContext::getDefault($context));
+                }
+            }
+        }
+
         // Roll the dice for sampling transaction, all child spans inherit the sampling decision.
-        if (null === $context->sampled) {
+        // And only if $$sampleRate == null because then the traces_sampler wasn't defined and we need to roll the dice
+        if (null === $context->sampled && null == $sampleRate) {
             if (null !== $client) {
                 $sampleRate = $client->getOptions()->getTracesSampleRate();
-
-                if ($sampleRate < 1 && mt_rand(1, 100) / 100.0 > $sampleRate) {
-                    // if true = we want to have the transaction
-                    // if false = we don't want to have it
-                    $context->sampled = false;
-                } else {
-                    $context->sampled = true;
-                }
             }
         }
 
+        if ($sampleRate < 1 && mt_rand(1, 100) / 100.0 > $sampleRate) {
+            // if true = we want to have the transaction
```

Comment:
I think that these comments are unnecessary, to me the code is pretty self-explanatory, isn't it?